### PR TITLE
Feature/replace portal by connection

### PR DESCRIPTION
**Pull request by whitx**
# Description
It is time to move out from portal profile to connection profile

# Impacts
Portal, portal profile, connection

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Portal Profile has been renamed as Connection Profile

# UPGRADE file entries

Admin Roles Configuration
^^^^^^^^^^^^^^^^^^^^^^^^^
You must rename "PORTAL_PROFILE" to "CONNECTION_PROFILE" in your adminroles.conf configuration file and reload the configuration after. You can automate this using:

  cd /usr/local/pf
  sed -i "s/PORTAL_PROFILE/CONNECTION_PROFILE/g" conf/adminroles.conf
  bin/pfcmd configreload hard

### CODE REVIEW

**Comment by whitx** [comment]
Tested the flow with 2 different connection profile, the user is properly send to the wanted connection profile.

File: lib/pf/Portal/Profile.pm [hunk 1]
```diff
@@ -69,7 +69,7 @@ sub new {
```
Comment (julsemaan) [inline lib/pf/Portal/Profile.pm:69]: I was expecting this file to be renamed as well to lib/pf/Connection/Profile.pm and the class name to be changed

File: lib/pf/Portal/Session.pm [hunk 2]
```diff
@@ -449,7 +449,7 @@ sub setDestinationUrl {
 
 =item getProfile
 
-Returns the proper captive portal profile for the current session.
+Returns the proper captive connection profile for the current session.
```
Comment (julsemaan) [inline lib/pf/Portal/Session.pm]: captive connection profile->connection profile

File: lib/pf/Portal/Profile.pm (hunk 1 above)
Comment (whitx) [reply lib/pf/Portal/Profile.pm:69]: Alright make sense, working on it

**Comment by extrafu** [comment]
Reybaisse

**Comment by whitx** [comment]
rebaissed

**Comment by extrafu** [comment]
@julsemaan and @dwlfrth BUMP! This needs to be merged ASAP to get good testing and easy other PRs.

**Comment by dwlfrth** [comment]
Not quite sure the "Enhancement" section is really an ... enhancement ?
Also, missing UPGRADE notes which seems to be mandatory due to configuration file change

File: docs/PacketFence_Cisco_MSE_Quick_Install_Guide.asciidoc [hunk 3]
```diff
@@ -49,16 +49,16 @@ So now each time a device will enter or leave the specific zone PacketFence will
 Step 3: Configure a portal filter
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-In `Configuration -> Portal Profiles -> Add profile -> Advanced filter` specify an advanced filter like that:
+In `Configuration -> Connection Profiles -> Add profile -> Advanced filter` specify an advanced filter like that:
 
  extended.mse_inout.locationMapHierarchy == "Campus>Building>Level>Zone"
 
 image::docs/images/Portal_filter.png[scaledwidth="100%",alt="Portal Filter"]
```
Comment (dwlfrth) [inline docs/PacketFence_Cisco_MSE_Quick_Install_Guide.asciidoc]: Should we adjust the screenshots ? (and image path / alt text)

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 4]
```diff
@@ -4318,7 +4318,7 @@ image::docs/images/aruba-l3-auth2.png[scaledwidth="100%",alt="Layer 3 profile 2"
 
 Now create the policy for the guest access, for example Internet only.
 
-Add the authentication for the Captive Portal Profile via `Security -> Authentication -> L3 Authentication -> Captive Portal Authentication Profile -> Server Group`:
```
Comment (dwlfrth) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Nope !

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 5]
```diff
@@ -4365,7 +4365,7 @@ Source NAT per Application
   alias "user" alias "packetfence-portal" "svc-http" src-nat queue low
   alias "user" alias "packetfence-portal" "svc-https" src-nat queue low
 
-Now add the newly created policy to the Captive Portal Profile, `Security -> User Roles`:
```
Comment (dwlfrth) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Nope !

File: html/captive-portal/lib/captiveportal.pm [hunk 6]
```diff
@@ -190,7 +190,7 @@ has portalSession => (
 
 sub _build_portalSession {
     my ($c) = @_;
-    return $c->model('Portal::Session');
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal.pm]: Nope !

File: html/captive-portal/lib/captiveportal/Model/Connection/Session.pm [hunk 7]
```diff
@@ -1,11 +1,11 @@
-package captiveportal::Model::Portal::Session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/Model/Connection/Session.pm [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
+package captiveportal::Model::Connection::Session;
 use Moose;
 
-extends 'captiveportal::PacketFence::Model::Portal::Session';
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/Model/Connection/Session.pm [hunk 9]
```diff
@@ -1,11 +1,11 @@
...
+extends 'captiveportal::PacketFence::Model::Connection::Session';
 
 =head1 NAME
 
-captiveportal::Model::Portal::Session - Catalyst Model
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Email.pm [hunk 10]
```diff
@@ -10,7 +10,7 @@ use pf::constants;
 use pf::config qw(%Config);
 use pf::activation qw($GUEST_ACTIVATION $SPONSOR_ACTIVATION);
 use pf::node;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Email.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Controller/Root.pm [hunk 11]
```diff
@@ -2,15 +2,15 @@ package captiveportal::PacketFence::Controller::Root;
...
-use pf::constants::Portal::Profile qw($PENDING_POLICY);
+use pf::constants::Connection::Profile qw($PENDING_POLICY);
 use URI::Escape::XS qw(uri_escape uri_unescape);
 use HTML::Entities;
 use pf::enforcement qw(reevaluate_access);
 use pf::config qw(%Config $fqdn);
 use pf::file_paths qw($conf_dir);
 use pf::log;
 use pf::util;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Controller/Root.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm [hunk 12]
```diff
@@ -34,7 +34,7 @@ use pf::util;
 use pf::violation;
 use pf::constants::scan qw($POST_SCAN_VID);
 use pf::inline;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm [hunk 13]
```diff
@@ -115,7 +115,7 @@ sub handle_web_form_release {
             $switch = pf::SwitchFactory->instantiate($last_switch_id);
         }
     }
-    my $session = new pf::Portal::Session()->session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm [hunk 14]
```diff
@@ -1,4 +1,4 @@
-package captiveportal::PacketFence::Model::Portal::Session;
+package captiveportal::PacketFence::Model::Connection::Session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm [hunk 15]
```diff
@@ -30,7 +30,7 @@ use List::MoreUtils qw(any);
 
 =head1 NAME
 
-captiveportal::PacketFence::Model::Portal::Session - Catalyst Model
+captiveportal::PacketFence::Model::Connection::Session - Catalyst Model
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm [hunk 16]
```diff
@@ -246,12 +246,12 @@ sub _build_profile {
     my ($self) = @_;
     my $options =  $self->options;
     $options->{'last_ip'} = $self->clientIP->normalizedIP;
-    return pf::Portal::ProfileFactory->instantiate( $self->clientMac, $options );
+    return pf::Connection::ProfileFactory->instantiate( $self->clientMac, $options );
 }
 
 sub _build_dispatcherSession {
     my ($self) = @_;
-    my $session = new pf::Portal::Session()->session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm]: Nope !

File: html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm [hunk 17]
```diff
@@ -246,12 +246,12 @@ sub _build_profile {
...
+    my $session = new pf::Connection::Session()->session;
```
Comment (dwlfrth) [inline html/captive-portal/lib/captiveportal/PacketFence/Model/Connection/Session.pm]: Nope !

File: lib/pf/Connection/Session.pm [hunk 18]
```diff
@@ -1,14 +1,14 @@
-package pf::Portal::Session;
+package pf::Connection::Session;
```
Comment (dwlfrth) [inline lib/pf/Connection/Session.pm]: Nope !

File: lib/pf/Connection/Session.pm [hunk 19]
```diff
@@ -1,14 +1,14 @@
...
 
 =head1 NAME
 
-pf::Portal::Session
+pf::Connection::Session
```
Comment (dwlfrth) [inline lib/pf/Connection/Session.pm]: Nope !

File: lib/pf/Connection/Session.pm [hunk 20]
```diff
@@ -1,14 +1,14 @@
...
 
 =cut
 
 =head1 DESCRIPTION
 
-pf::Portal::Session wraps several parameter we often need from the captive
+pf::Connection::Session wraps several parameter we often need from the captive
```
Comment (dwlfrth) [inline lib/pf/Connection/Session.pm]: Nope !

File: lib/pf/web/dispatcher.pm [hunk 21]
```diff
@@ -39,7 +39,7 @@ use pf::web::constants;
 use pf::web::filter;
 use pf::web::util;
 use pf::proxypassthrough::constants;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline lib/pf/web/dispatcher.pm]: Nope !

File: lib/pf/web/externalportal.pm [hunk 22]
```diff
@@ -29,7 +29,7 @@ use pf::config qw(
 use pf::ip4log;
 use pf::log;
 use pf::locationlog qw(locationlog_view_open_mac locationlog_get_session);
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline lib/pf/web/externalportal.pm]: Nope !

File: lib/pf/web/externalportal.pm [hunk 23]
```diff
@@ -154,7 +154,7 @@ sub _setup_session {
     my %info = (
         'client_mac' => $client_mac,
     );
-    my $portalSession = pf::Portal::Session->new(%info);
```
Comment (dwlfrth) [inline lib/pf/web/externalportal.pm]: Nope !

File: lib/pf/web/release.pm [hunk 24]
```diff
@@ -23,7 +23,7 @@ use pf::config;
 use pf::file_paths qw($bin_dir);
 use pf::ip4log;
 use pf::node;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline lib/pf/web/release.pm]: Nope !

File: lib/pf/web/release.pm [hunk 25]
```diff
@@ -38,7 +38,7 @@ sub handler
 
   my $logger = get_logger();
 
-  my $portalSession     = pf::Portal::Session->new();
```
Comment (dwlfrth) [inline lib/pf/web/release.pm]: Nope !

File: lib/pf/web/wispr.pm [hunk 26]
```diff
@@ -35,7 +35,7 @@ use pf::config qw(
 use pf::ip4log;
 use pf::node;
 use pf::web;
-use pf::Portal::Session;
```
Comment (dwlfrth) [inline lib/pf/web/wispr.pm]: Nope !

File: lib/pf/web/wispr.pm [hunk 27]
```diff
@@ -59,7 +59,7 @@ sub handler {
 
     $logger->trace("hitting wispr");
 
-    my $portalSession = pf::Portal::Session->new();
```
Comment (dwlfrth) [inline lib/pf/web/wispr.pm]: Nope !

**Review by dwlfrth — CHANGES_REQUESTED** [review]
Not sure we want to move Portal::Session to Connection::Session

Also, is everything working fine following the changes ?

**Comment by extrafu** [comment]
Rebase

**Review by dwlfrth — APPROVED** [review]

**Pull request merged**